### PR TITLE
chore: get rid of kv_args and replace it with slices to full_args

### PR DESCRIPTION
**Pull request by romange**
Fixing the performance bug caused by applying equality operator
on two spans inside ShardArgs::Iterator::operator==

### CODE REVIEW

File: src/server/stream_family.cc [hunk 1]
```diff
@@ -3077,12 +3077,15 @@ void XReadImpl(CmdArgList args, ReadOpts* opts, ConnectionContext* cntx) {
   for (size_t i = 0; i < xread_resp.size(); i++) {
     ShardId sid = tx->GetUniqueShardCnt() > 1 ? i : tx->GetUniqueShard();
     vector<RecordVec> sub_results = std::move(xread_resp[i]);
+
     ShardArgs shard_args = cntx->transaction->GetShardArgs(sid);
-    for (size_t j = 0; j < sub_results.size(); j++) {
+    auto shard_args_it = shard_args.begin();
```
Comment (romange) [inline src/server/stream_family.cc:3082]: @dranikpg should `DCHECK_EQ(shard_args.Size(), sub_results.size());` hold? because it does not.

**Comment by romange** [comment]
@dranikpg I fixed the bug that caused the slowdown. it was in the iterator equality operator. Please review the PR

**Comment by dranikpg** [comment]
> @dranikpg I fixed the bug that caused the slowdown. it was in the iterator equality operator. Please review the PR

Will do. Please stop foce pushing so I could've actually seen what the bug war 🙂

**Comment by dranikpg** [comment]
Didn't work 😅 Will take a detailed look

File: src/server/command_registry.h [hunk 2]
```diff
@@ -27,16 +27,13 @@ enum CommandOpt : uint32_t {
   LOADING = 1U << 3,  // Command allowed during LOADING state.
   DENYOOM = 1U << 4,  // use-memory in redis.
 
-  // marked commands that demand preserve the order of keys to work correctly.
-  // For example, MGET needs to know the order of keys to return the values in the same order.
-  // BLPOP needs to know the order of keys to return the first non-empty list from the left.
-  REVERSE_MAPPING = 1U << 5,
+  // UNUSED = 1U << 5,
```
Comment (dranikpg) [inline src/server/command_registry.h:30]: they're not stored anywhere, we can shift them freely

File: src/server/list_family.cc [hunk 3]
```diff
@@ -158,6 +158,22 @@ struct CircularMessages {
...
+// A bit awkward translation from a single key to ShardArgs.
+// We create a mutable slice (which will never be mutated) from the key, then we create
+// a CmdArgList of size 1 that references mslice and finally
+// we reference the first element in the CmdArgList via islice.
+struct SingleArg {
+  MutableSlice mslice;
+  IndexSlice islice{0, 1};
+
+  SingleArg(string_view arg) : mslice(const_cast<char*>(arg.data()), arg.size()) {
+  }
```
Comment (dranikpg) [inline src/server/list_family.cc:170]: that's tricky and only highlights to our bigger problem of exposing too much mutable slices

All `FinalizeWatched` does with args is this `for (string_view key : args) {`

**Review by dranikpg — APPROVED** [review]

File: src/server/list_family.cc (hunk 3 above)
Comment (romange) [reply src/server/list_family.cc:170]: if I remember correctly `FinalizeWatched` is tricky because of BRPOPLPUSH

**Pull request merged**